src/x/checked/finalizeable_once.go

```diff
@@ -0,0 +1,18 @@
+package checked
+
+import "go.uber.org/atomic"
+
+// FinalizeableOnce implements the logics needed to enforce single Finalize on pool.ObjectPool.
+type FinalizeableOnce struct {
```

Comment:
We should probably embed this to `RefCount` struct and get this checking for all things that embed `RefCount`.

Reply:
Already a part of this PR.